lib/apartment/adapters/mysql2_adapter.rb

```diff
@@ -32,6 +32,19 @@ def current_database_name=(name)
         Thread.current[:apartment_current_database_name] = name
       end
 
+      # Override process method to ensure current database is restored propertly
+      def process(database_config = nil, use_use=true)
+        saved_config = current_database.merge({target_database: current_database_name})
```

Comment:
Will `current_database` always exist the first time this method is invoked? Should we add a `try` block on the merge method?